test/widgets/compose_box_test.dart

```diff
@@ -573,97 +600,199 @@ void main() {
+          final deactivatedUsers = [eg.user(isActive: false), eg.user(isActive: false)];
+          await prepareComposeBox(tester, narrow: groupDmNarrowWith(deactivatedUsers),
+            users: deactivatedUsers);
+          checkComposeBox(isShown: false);
+
+          await changeUserStatus(tester, user: deactivatedUsers[0], isActive: true);
+          checkComposeBox(isShown: false);
+
+          await changeUserStatus(tester, user: deactivatedUsers[1], isActive: true);
+          checkComposeBox(isShown: true);
+        });
       });
+    });
 
-      testWidgets('deactivated user becomes active -> '
-          'banner is replaced with the compose box', (tester) async {
-        final deactivatedUser = eg.user(isActive: false);
-        await prepareComposeBox(tester, narrow: dmNarrowWith(deactivatedUser),
-          users: [deactivatedUser]);
-        checkComposeBox(isShown: false);
+    group('in channel/topic narrow according to channel post policy', () {
+      void checkComposeBox({required bool isShown}) => checkComposeBoxIsShown(isShown,
+        bannerLabel: zulipLocalizations.errorBannerCannotPostInChannelLabel);
+
+      final narrowTestCases = [
+        ('channel', const ChannelNarrow(1)),
+        ('topic',   const TopicNarrow(1, 'topic')),
+      ];
+
+      for (final (String narrowType, Narrow narrow) in narrowTestCases) {
+        testWidgets('compose box is shown in $narrowType narrow', (tester) async {
+          await prepareComposeBox(tester,
+            narrow: narrow,
+            selfUser: eg.user(role: UserRole.administrator),
+            streams: [eg.stream(streamId: 1,
+              channelPostPolicy: ChannelPostPolicy.moderators)]);
+          checkComposeBox(isShown: true);
+        });
+
+        testWidgets('error banner is shown in $narrowType narrow', (tester) async {
+          await prepareComposeBox(tester,
+            narrow: narrow,
+            selfUser: eg.user(role: UserRole.moderator),
+            streams: [eg.stream(streamId: 1,
+              channelPostPolicy: ChannelPostPolicy.administrators)]);
+          checkComposeBox(isShown: false);
+        });
+      }
 
-        await changeUserStatus(tester, user: deactivatedUser, isActive: true);
+      Future<void> changeUserRole(WidgetTester tester, {
+        required User user,
+        required UserRole role,
+      }) async {
+        await store.handleEvent(RealmUserUpdateEvent(id: 1,
+          userId: user.userId, role: role));
+        await tester.pump();
+      }
+
+      Future<void> changeChannelPolicy(WidgetTester tester, {
+        required ZulipStream channel,
+        required ChannelPostPolicy policy,
+      }) async {
+        await store.handleEvent(eg.channelUpdateEvent(channel,
+          property: ChannelPropertyName.channelPostPolicy, value: policy));
+        await tester.pump();
+      }
+
+      testWidgets('user role decreases -> compose box is replaced with the banner', (tester) async {
```

Comment:
I am not sure if the verbs "decrease" and "increase" are the proper ones to use here! 😃 I thought a lot about them, but couldn't come up with better ones. Any suggestions would be appreciated!